### PR TITLE
[HttpKernel] Update DebugLoggerInterface::getLogs PHPDoc

### LINKED ISSUE #47396: [HttpKernel] Outdated DebugLoggerInterface::getLogs PHPDoc
### Symfony version(s) affected

5.4, 6

### Description

While trying to implement a `DebugLoggerInterface` around HttpKernel’s `Logger` I read the following:

https://github.com/symfony/symfony/blob/cdd1c08bf862055a361428293c1f6237ce572569/src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php#L26-L28

But it seems that since https://github.com/symfony/symfony/pull/42195 the `LoggerDataCollector` will expect the keys

- `channel`
- `context`
- `message`
- `priority`
- `priorityName`
- `timestamp_rfc3339`

So

- `timestamp` is not needed
- `context` is not optional
- `timestamp_rfc3339`and `channel` are mandatory

It seems to me `DebugLoggerInterface` is almost exclusively used by the `LoggerDataCollector` so I guess the keys should match.

### How to reproduce

Decorate HttpKernel’s logger to implement `DebugLoggerInterface`:

```php
<?php

namespace App\Log;

use Psr\Log\LoggerInterface;
use Psr\Log\LoggerTrait;
use Psr\Log\LogLevel;
use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\RequestStack;
use Symfony\Component\HttpKernel\Log\DebugLoggerInterface;

#[AsDecorator('logger')]
class DebugLogger implements LoggerInterface, DebugLoggerInterface
{
    use LoggerTrait;
    
    private array $logsByRequest = [];
    
    public function __construct(
        private readonly LoggerInterface $logger,
        private readonly RequestStack $requestStack,
    ) {}

    public function log($level, $message, array $context = []): void
    {
        $this->logger->log($level, $message, $context);
        
        $request = $this->requestStack->getCurrentRequest();

        $this->logsByRequest[$request ? spl_object_hash($request) : ''][] = [
            'context' => $context,
            'message' => $message,
            'priority' => [
                LogLevel::DEBUG => 0,
                LogLevel::INFO => 1,
                LogLevel::NOTICE => 2,
                LogLevel::WARNING => 3,
                LogLevel::ERROR => 4,
                LogLevel::CRITICAL => 5,
                LogLevel::ALERT => 6,
                LogLevel::EMERGENCY => 7,
            ][$level],
            'priorityName' => $level,
            'timestamp' => time(),
        ];
    }

    public function getLogs(Request $request = null): array
    {
        if ($request) {
            return $this->logsByRequest[spl_object_hash($request)] ?? [];
        }
        
        return array_merge(...array_values($this->logsByRequest));
    }

    public function countErrors(Request $request = null): int
    {
        $errorCount = 0;
        foreach ($this->getLogs($request) as $log) {
            if ($log['priority'] >= 4) {
                ++$errorCount;
            }
        }
        return $errorCount;
    }

    public function clear()
    {
        $this->logsByRequest = [];
    }
}

```

then access the profiler’s logger panel. It will crash.

### Possible Solution

Update mandatory keys to be used by the `LoggerDataCollector`.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #47396 
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]
(for 6.2 since this is not a bugfix)

**Review by derrabus — APPROVED** [review]

**Comment by MatTheCat** [comment]
Rebased on 6.2 :ok_hand:

File: src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php [hunk 1]
```diff
@@ -23,11 +23,15 @@ interface DebugLoggerInterface
     /**
      * Returns an array of logs.
      *
-     * A log is an array with the following mandatory keys:
-     * timestamp, message, priority, and priorityName.
-     * It can also have an optional context key containing an array.
-     *
-     * @return array
+     * @return array<array{
+     *     channel: ?string,
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php]: is the key mandatory but nullable or is it optional ?

File: src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php [hunk 2]
```diff
@@ -23,11 +23,15 @@ interface DebugLoggerInterface
...
+     *     context: array,
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php]: I would use `array<string, mixed>`. We expect it to be an associative array.

File: src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php (hunk 1 above)
Comment (MatTheCat) [reply src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php]: Mandatory key, nullable value.

File: src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php [hunk 3]
```diff
@@ -23,11 +23,15 @@ interface DebugLoggerInterface
...
      * Returns an array of logs.
      *
-     * A log is an array with the following mandatory keys:
-     * timestamp, message, priority, and priorityName.
-     * It can also have an optional context key containing an array.
-     *
-     * @return array
+     * @return array<array{
+     *     channel: ?string,
+     *     context: array<string, mixed>,
```
Comment (OskarStark) [inline src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php]: Could be an empty array too

Comment (MatTheCat) [reply src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php]: Is there a specific annotation for empty arrays?

**Review by maxhelias — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php (hunk 3 above)
Comment (stof) [reply src/Symfony/Component/HttpKernel/Log/DebugLoggerInterface.php]: An empty array totally satisfies `array<string, mixed>`

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @MatTheCat.

**Pull request closed without merging**